script.js

```diff
@@ -35,25 +35,43 @@
     TOMApp.prototype.render = function() {
       var self = this;
 
+      // reset the region
       self.regions.$mainRegion.empty();
+      $(".action-remove").off("click"); // prevent memory leaking
 
+      // rendering
       var $row = $("<div>").addClass("row");
-      for(var i=0; i< self.products.length ; i++){
+      for(var i=0; i < self.products.length ; i++){
         $row.append(self.products[i].render());
       }
       self.regions.$mainRegion.append($row);
+
+      // attach event handlers for removing product button 
+      $(".action-remove").on("click", function(){
+        event.preventDefault();
+
+        var $container = $(this).closest(".product-container");
+
+        var self = this;
+        $container.on("animationend", function() {
```

Comment:
When the animation has ended, remove the jQuery DOM and turn off event handler to address memory leak.